src/Zio/FileSystems/ZipArchiveFileSystem.cs

```diff
@@ -934,29 +938,33 @@ public ZipEntryStream(FileShare share, ZipArchiveFileSystem system, ZipArchiveEn
             {
                 _entry = entry;
                 _fileSystem = system;
-                var fileShare = _fileSystem._openStreams.TryGetValue(entry, out var fileData) ? fileData.Share : FileShare.ReadWrite;
-                if (fileData != null)
+
+                lock (_fileSystem._openStreamsLock)
```

Comment:
just found out about concurrent dictionary, think it'll be better here 🙂

Reply:
it's only avaliable in standard 2.1, eh

Reply:
I think using ConcurrentDictionary without locks is actually impossible, as`entry.Open()` throws an exception when accessed by two threads at the same time. I would go with locks for now